.github/workflows/build.yml

```diff
@@ -0,0 +1,43 @@
+# The present workflow was made based on the following references:
+# - https://github.com/actions/cache/blob/main/examples.md#haskell---cabal
+# - https://github.com/haskell/time/blob/master/.github/workflows/ci.yml
+# - https://github.com/stackbuilders/stache/blob/master/.github/workflows/ci.yaml
+# - https://markkarpov.com/post/github-actions-for-haskell-ci.html
+---
+name: Build
+
+on: push
+
+concurrency:
+  group: build-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  haskell:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        ghc: ['9.0', '8.10', '8.6.1', '8.4.3', '8.2']
```

Comment:
Shouldn't this include the most recent versions of GHC, '9.4' and '9.2'?
Also, here I'd drop the minor version, so just `8.6` and `8.4`, unless you are especially interested in the superseded versions `8.6.1` (latest is `8.6.5`) and `8.4.3` (latest is `8.4`), but I would wonder why so.

Reply:
@andreasabel Thanks, I will set the GHC versions to each major version without specifying the revision so the build process is tested against the latest revision for each version:
In other words:
```
      matrix:
        ghc: ['8.2', '8.4', '8.6', '8.8', '8.10', '9.0']

```